### PR TITLE
WIP: 6.0 version change file name hash function from md5 to base64 safe url

### PR DESCRIPTION
**Pull request by Insofan**
### New Pull Request Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [x] I have searched for a similar pull request in the [project](https://github.com/rs/SDWebImage/pulls) and found none

* [x] I have updated this branch with the latest master to avoid conflicts (via merge from master or rebase)
* [ ] I have added the required tests to prove the fix/feature I am adding
* [ ] I have updated the documentation (if necessary)
* [x] I have run the tests and they pass
* [x] I have run the lint and it passes (`pod lib lint`)

This merge request fixes / refers to the following issues: ...

### Pull Request Description
Change file name hash function from md5 to sha1 ([reference issue](https://github.com/SDWebImage/SDWebImage/issues/3018)). Has run test31CachePathForAnyKey and test32CachePathForNilKey.

### CODE REVIEW

**Comment by jonthanon** [comment]
SHA1 was broken too in 2017, so unfortunately it's insecure as well. The recommendation I'm seeing around is to go to SHA256.

**Comment by Insofan** [comment]
> SHA1 was broken too in 2017, so unfortunately it's insecure as well. The recommendation I'm seeing around is to go to SHA256.

From my opinion encrypt file name didn't protect any thing, maybe just helpful for some company's security test. It only just  a hash function for generate disk filename, not any cryptographic.

File: SDWebImage/Core/SDDiskCache.m [hunk 1]
```diff
@@ -308,17 +308,17 @@ - (void)moveCacheDirectoryFromPath:(nonnull NSString *)srcPath toPath:(nonnull N
     if (str == NULL) {
         str = "";
     }
-    unsigned char r[CC_MD5_DIGEST_LENGTH];
-    CC_MD5(str, (CC_LONG)strlen(str), r);
+    unsigned char r[CC_SHA256_DIGEST_LENGTH];
+    CC_SHA1(str, (CC_LONG)strlen(str), r);
```
Comment (Legonaftik) [inline SDWebImage/Core/SDDiskCache.m:312]: Why `CC_SHA1` here instead of `CC_SHA256`? Looks like a typo.

**Review by Legonaftik — APPROVED** [review]

File: SDWebImage/Core/SDDiskCache.m (hunk 1 above)
Comment (Insofan) [reply SDWebImage/Core/SDDiskCache.m:312]: This pr maybe discard.

**Pull request closed without merging**